libyul/backends/evm/EVMCodeTransform.cpp

```diff
@@ -628,13 +655,72 @@ void CodeTransform::visitExpression(Expression const& _expression)
 	expectDeposit(1, height);
 }
 
+void CodeTransform::setupReturnVariablesAndFunctionExit()
+{
+	yulAssert(!m_functionExitStackHeight.has_value(), "");
+	yulAssert(m_scope, "");
+
+	ScopeGuard scopeGuard([oldScope = m_scope, this] { m_scope = oldScope; });
+	if (!m_scope->functionScope)
+	{
+		yulAssert(m_scope->superScope && m_scope->superScope->functionScope, "");
+		m_scope = m_scope->superScope;
+	}
+
+	if (m_delayedReturnVariables.empty())
+	{
+		m_functionExitStackHeight = m_evm15 ? 0 : 1;
+		return;
+	}
+
+	// Allocate slots for return variables as if they were declared as variables in the virtual function scope.
+	for (TypedName const& var: m_delayedReturnVariables)
+		(*this)(VariableDeclaration{var.location, {var}, {}});
+
+	m_functionExitStackHeight = ranges::max(m_delayedReturnVariables | ranges::views::transform([&](TypedName const& _name) {
+		return variableStackHeight(_name.name);
+	})) + 1;
```

Comment:
```suggestion
	})) + (m_evm15 ? 0 : 1);
```
?

Reply:
Nope - that part is already implicit in the stack height of the return variables itself (there will be at least one of them here).

Example:
If there is one return variable and ``m_evm15`` is true, then that return variable will live at slot `0` and the function exit stack height will be `0+1 = 1`.
If there is one return variable and ``m_evm15`` is false, then that return variable will live at alot `1` and the function exit statck height will be `1+1 = 2`.
etc.

Reply:
ok!